pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_status.go

```diff
@@ -0,0 +1,106 @@
+package gatewayapi
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kube_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kube_schema "k8s.io/apimachinery/pkg/runtime/schema"
+	kube_types "k8s.io/apimachinery/pkg/types"
+	kube_client "sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+func (r *HTTPRouteReconciler) findReference(gk kube_schema.GroupKind) (kube_client.Object, error) {
+	versions := r.Scheme.VersionsForGroupKind(
+		kube_schema.GroupKind{Group: gk.Group, Kind: gk.Kind},
+	)
+
+	if len(versions) == 0 {
+		return nil, nil
+	}
+
+	gvk := kube_schema.GroupVersionKind{
+		Group:   gk.Group,
+		Kind:    gk.Kind,
+		Version: versions[0].Version,
+	}
+
+	rawObj, err := r.Scheme.New(gvk)
+	if err != nil {
+		return nil, err
+	}
+
+	obj, ok := rawObj.(kube_client.Object)
+	if !ok {
+		return nil, nil
+	}
+
+	return obj, nil
+}
+
+func (r *HTTPRouteReconciler) resolvedRefCondition(
+	ctx context.Context, route kube_client.Object, ref gatewayapi.ParentRef,
+) (kube_meta.Condition, error) {
+	namespace := route.GetNamespace()
+	if ns := ref.Namespace; ns != nil {
+		namespace = string(*ns)
+	}
+
+	resolvedStatus := kube_meta.ConditionTrue
+	resolvedReason := gatewayapi.ConditionRouteResolvedRefs
+
+	obj, err := r.findReference(kube_schema.GroupKind{Group: string(*ref.Group), Kind: string(*ref.Kind)})
```

Comment:
I'm confused about why we want to make this generic. We only support Gateways being the parents of routes, so I would expect the logic to be to start with a route, follow the ref to the parent Gateway, then check the gateway class on that to see whether we should process it.

Reply:
Indeed, we need to check the gateway class, I'll add that.

Reply:
All rewritten